### ISSUE TITLE
Incomplete certificate chain returned by custom DNS Suffix setup

### ISSUE THREAD

**Issue by Kalmis** [microsoft/azure-container-apps#504]
### This issue is a: (mark with an x)
- [x] bug report -> please search issues before submitting
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)

### Issue description
After setting up a custom domain suffix for Container Apps Environment with full certificate chain (leaf + intermediate + root + key) then all container apps are still returning only the leaf certificate. See screenshot below from SSL Labs test. I tried with the following formats and chains. Also, tried waiting some hours + creating new revisions for container apps.
PEM (no password)
- leaf + intermediate + key
- leaf + intermediate + root + key
PFX
- leaf + intermediate + root + key

### Steps to reproduce
1. Create a PEM/PFX with the full certificate chain
2. Navigate to your Container Apps Engine -> Custom DNS Suffix
3. Do the DNS custom setup + upload your full certificate chain

**Expected behavior**
Container apps return the full chain which was uploaded.

**Actual behavior**
Container apps return only the leaf certificate.

**Screenshots**         
<img width="1011" alt="Screenshot 2022-11-22 at 9 51 17" src="https://user-images.githubusercontent.com/19410121/203256476-dc43042b-ea82-4f01-a1e5-9e0a803a5a90.png">

### Additional context
Testing via the Portal. Also, the subscription does not have support plan.

**Comment by Seris370** [comment]
Hi Kalmis, we are investigating the issue, but have not managed to reproduce the incomplete chain. Could you provide the following information to help us proceed with the investigation?
1. Could you share the environment information (e.g. you subscription id, resource group name, environment name) to our email [Azure Container Apps Discussions](mailto:acadisc@microsoft.com)?
2. Could you open the uploaded PEM file in plain text and verify that it contains three sections that start with "-----BEGIN CERTIFICATE-----" and end with "-----END CERTIFICATE-----"?
3. Are there self-signed certificates in the certificate chain?

**Comment by Kalmis** [comment]
Hello, thanks for the investigation! I got it working now. I uploaded a wrong certificate chain (for another domain) to be sure that the certificate is updated. Then I uploaded the correct chain in PEM format (order leaf + intermediate + root + key), and it seems to work now.

What comes to the questions,
2. The PEM file contains four sections. Three begin certificate sections (leaf, intermediate and root certificates) + the begin private section where the key is.
3. No self-signed certificates were used.
